### PR TITLE
add Mako to templating section

### PR DESCRIPTION
**Pull request by SanketDG**
http://www.makotemplates.org/

Mako is the default templating language included in the pyramid
and pylons web frameworks. This includes a short info on how to
get started writing templates in Mako.

_I am not sure what to keep and what to omit._

### CODE REVIEW

File: docs/scenarios/web.rst [hunk 1]
```diff
@@ -473,6 +473,45 @@ you can replace it with a more terse and readable syntax that uses the pattern
...
 
+Mako
+----
+`Mako <http://www.makotemplates.org/>`_ is a template language that compiles to Python
+for maximum performance. Its syntax and api is borrowed from the best parts of other
+templating languages like Django and Jinja2 templates. It is the default template
+language included with the `Pylons and Pyramid <http://www.pylonsproject.org/>`_ web
+frameworks.
+
+Mako is used by `reddit.com <http://www.reddit.com>`_.
```
Comment (sigmavirus24) [inline docs/scenarios/web.rst]: We don't list corporate users for other libraries or templateing engines here. Please remove this.

File: docs/scenarios/web.rst [hunk 2]
```diff
@@ -473,6 +473,45 @@ you can replace it with a more terse and readable syntax that uses the pattern
...
+
+The template language is simple enough:
```
Comment (sigmavirus24) [inline docs/scenarios/web.rst]: This is antagonistic towards people new to the library. Reword this to:

```
An example template looks like:
```

File: docs/scenarios/web.rst [hunk 3]
```diff
@@ -473,6 +473,45 @@ you can replace it with a more terse and readable syntax that uses the pattern
...
+
+.. code-block:: html
+
+  <%inherit file="base.html"/>
```
Comment (sigmavirus24) [inline docs/scenarios/web.rst]: This is inconsistently indented with the rest of the code-samples, including the next one. Please use a 4 space indent for this block of code.

File: docs/scenarios/web.rst [hunk 4]
```diff
@@ -473,6 +473,45 @@ you can replace it with a more terse and readable syntax that uses the pattern
...
+
+  <%def name="makerow(row)">
+      <tr>
+      % for name in row:
+          <td>${name}</td>\
+      % endfor
+      </tr>
+  </%def>
+
+Mako has a simple API for basic usage:
```
Comment (sigmavirus24) [inline docs/scenarios/web.rst]: Again, calling things "simple" is antagonistic. This should read:

```
To then render this template, you can do the following:
```

**Comment by SanketDG** [comment]
@sigmavirus24 Does this look okay now?
